docs/threads.rst

```diff
@@ -0,0 +1,35 @@
+Multi-threading
+===============
+libhdf5, at least with the default compilation flags, is not thread-safe.
+
+h5py supports multi-threading, in the sense that you can use h5py from multiple threads
+without taking any special precautions. This is enabled by wrapping all critical
+sections that invoke the libhdf5 C API with an internal *interpreter-wide reentrant
+lock*, to be used either as a context manager or as a function decorator:
+
+.. code-block:: python
+
+    from h5py._objects import phil, with_phil
```

Comment:
maybe more of a question for more senior maintainers but since it comes up a lot in discussions lately: why is this lock named Phil ? 😅

Reply:
It was named that when I took over from @andrewcollette .